platform-controller/src/main/java/org/hobbit/controller/data/NodeHardwareInformation.java

```diff
@@ -137,11 +142,34 @@ public Resource addToModel(Model model) {
     }
 
     private StmtIterator distinguishingProperties(Model model, Resource self) {
+
+        // Check for nulls and provide default values
+        String instanceValue = (instance == null) ? "default-instance" : instance;
+        if (instance == null) {
+            LOGGER.warn("instance is null, using default value: {}", instanceValue);
+        }
+
+        String cpuValue = (cpu == null) ? "default-cpu" : cpu;
+        if (cpu == null) {
+            LOGGER.warn("cpu is null, using default value: {}", cpuValue);
+        }
+
+        String memoryValue = (memory == null) ? "default-memory" : memory;
+        if (memory == null) {
+            LOGGER.warn("memory is null, using default value: {}", memoryValue);
+        }
+
+        String osValue = (os == null) ? "default-os" : os;
+        if (os == null) {
+            LOGGER.warn("os is null, using default value: {}", osValue);
+        }
+
+
         return new StmtIteratorImpl(
-                Stream.of((Statement) new StatementImpl(self, RDFS.label, model.createLiteral(instance)),
-                        (Statement) new StatementImpl(self, MEXCORE.cpu, model.createLiteral(cpu)),
-                        (Statement) new StatementImpl(self, MEXCORE.memory, model.createLiteral(memory)),
-                        (Statement) new StatementImpl(self, DOAP.os, model.createLiteral(os))).iterator());
+                Stream.of((Statement) new StatementImpl(self, RDFS.label, model.createLiteral(instanceValue)),
+                        (Statement) new StatementImpl(self, MEXCORE.cpu, model.createLiteral(cpuValue)),
+                        (Statement) new StatementImpl(self, MEXCORE.memory, model.createLiteral(memoryValue)),
+                        (Statement) new StatementImpl(self, DOAP.os, model.createLiteral(osValue))).iterator());
```

Comment:
It is a good idea to check for `null` values :+1: 

However, I am not sure whether it makes sense to insert `"null"` literals into the result graph :thinking: 
We should discuss that solution.